src/lib.rs

```diff
@@ -13,12 +18,48 @@ use std::sync::{ Arc, Mutex, MutexGuard };
 use std::sync::mpsc;
 
 #[cfg(windows)]
-fn get_default_device() -> Result<&'static str> {
-    "\\\\.\\COM6"
+fn get_default_device() -> Result<String> {
+    "\\\\.\\COM6".to_owned()
+}
+
+fn is_usb_zwave_device(port: &ListPortInfo) -> bool {
+    let default_usb_devices = [
+        // VID     PID
+        //-----   -----
+        (0x0658, 0x0200),   // Aeotech Z-Stick Gen-5
+    ];
+
+    // Is it one of the vid/pids in the table?
+    if let UsbPort(ref info) = port.port_type {
+        default_usb_devices.iter().find(|entry| entry.0 == info.vid && entry.1 == info.pid).is_some()
+    } else {
+        false
+    }
 }
 
 #[cfg(unix)]
-fn get_default_device() -> Result<&'static str> {
+fn get_default_device() -> Result<String> {
+
+    // Enumerate all of the serial devices and see if any of them match our
+    // known VID:PID.
+
+    if let Some(port) = ListPorts::new().iter().find(|port| is_usb_zwave_device(port)) {
```

Comment:
The HidController mode only exists to support older controllers which only supported HID mode instead of serial mode. When and if we get our hands on such a device, then we can add support for them. Until then we shouldn't bother.